csrc/scheduler/greedy.cpp

```diff
@@ -452,6 +590,78 @@ class ConstrainedOpScheduler : public OptOutDispatch {
     scheduleConstrainedTv(out_tv, {scan_dim});
   }
 
+  void handle(ScatterOp* scatter) override {
+    auto scatter_dim = scatter->dim();
+    auto in_tv = ir_utils::getTvInput(scatter);
+    auto index_tv = scatter->index()->as<TensorView>();
+    auto src_tv = scatter->src()->as<TensorView>();
+    auto out_tv = ir_utils::getTvOutput(scatter);
+
+    scheduleConstrainedTv(out_tv, {scatter_dim});
+    scheduleConstrainedTv(index_tv, {scatter_dim});
+    scheduleConstrainedTv(in_tv, {scatter_dim});
+    scheduleConstrainedTv(src_tv, {scatter_dim});
+
+    // If in_tv is produced by another scatter, it should be already
+    // scheduled, including its memory type. It should not be
+    // modified.
+    const bool in_tv_is_produced_by_scatter =
+        dynamic_cast<ScatterOp*>(in_tv->definition()) != nullptr;
```

Comment:
If input is coming from another scatter operation, didn't the previous `insertCopyAfterScatter` added another Load before its uses. So `in_tv_is_produced_by_scatter` is always false.

We should probably remove this as well as the true block below.